src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.AssemblyResolution.targets

```diff
@@ -168,6 +169,28 @@ _ResolveAssemblies MSBuild target.
         IncludeDebugSymbols="$(AndroidIncludeDebugSymbols)">
       <Output TaskParameter="OutputLibraries" ItemName="FrameworkNativeLibrary" />
     </ProcessNativeLibraries>
+    <ItemGroup>
+      <LocalFrameworkNativeLibrary Include="$(IntermediateOutputPath)native\%(FrameworkNativeLibrary.RuntimeIdentifier)\%(FrameworkNativeLibrary.Filename)%(FrameworkNativeLibrary.Extension)">
+        <RuntimeIdentifier>%(FrameworkNativeLibrary.RuntimeIdentifier)</RuntimeIdentifier>
+      </LocalFrameworkNativeLibrary>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_StripFrameworkNativeLibraries"
+      Condition=" '$(AndroidIncludeDebugSymbols)' != 'true' "
+      Inputs="@(FrameworkNativeLibrary)"
+      Outputs="@(LocalFrameworkNativeLibrary)">
+    <StripNativeLibraries
+        Libraries="@(FrameworkNativeLibrary)"
+        LocalPath="$(IntermediateOutputPath)native"
+        AndroidBinUtilsDirectory="$(AndroidBinUtilsDirectory)">
+      <Output TaskParameter="OutputLibraries" ItemName="StrippedFrameworkNativeLibrary" />
+    </StripNativeLibraries>
+    <ItemGroup Condition=" '$(AndroidIncludeDebugSymbols)' != 'true' ">
+      <FrameworkNativeLibrary Remove="@(FrameworkNativeLibrary)"/>
+      <FrameworkNativeLibrary Include="@(StrippedFrameworkNativeLibrary)"/>
+      <FileWrites Include="@(StrippedFrameworkNativeLibrary)" />
```

Comment:
Then I think these could also use `@(LocalFrameworkNativeLibrary)`, so we wouldn't need the `@(StrippedFrameworkNativeLibrary)` item group at all.

Reply:
The `@(LocalFrameworkNativeLibrary)` doesn't contain all the needed metadata information. So I think we would need the `@(StrippedFrameworkNativeLibrary)` anyway. OTOH, maybe I just don't have enough msbuild tricks at hand and there might be easy way to have it there, which I don't know?

Reply:
I pushed two commits, I hope that is OK. I think you just needed the `->` operator instead of batching with `%()`, to get the missing metadata.

I think it's working, I have a small `.apk` file that runs:
```
> 7z l .\bin\TestDebug\temp\DotNetBuildandroid-armTrue\bin\Release\net6.0-android\android-arm\UnnamedProject.UnnamedProject.apk

7-Zip 19.00 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2019-02-21

Scanning the drive for archives:
1 file, 3039199 bytes (2968 KiB)

Listing archive: .\bin\TestDebug\temp\DotNetBuildandroid-armTrue\bin\Release\net6.0-android\android-arm\UnnamedProject.UnnamedProject.apk

--
Path = .\bin\TestDebug\temp\DotNetBuildandroid-armTrue\bin\Release\net6.0-android\android-arm\UnnamedProject.UnnamedProject.apk
Type = zip
Physical Size = 3039199

   Date      Time    Attr         Size   Compressed  Name
------------------- ----- ------------ ------------  ------------------------
1980-12-31 18:00:00 .....         2584          887  AndroidManifest.xml
1980-12-31 18:00:00 .....         2199         2199  res\drawable-mdpi-v4\icon.png
1980-12-31 18:00:00 .....          544          268  res\layout\main.xml
1980-12-31 18:00:00 .....         1156         1156  resources.arsc
2021-02-23 16:48:20 .....       317288        72577  classes.dex
2021-02-23 16:48:20 .....         5662         5662  assemblies\UnnamedProject.dll
2021-02-23 16:48:20 .....         3584         3584  assemblies\es\UnnamedProject.resources.dll
2021-02-23 16:48:20 .....         6214         6214  assemblies\System.Console.dll
2021-02-23 16:48:20 .....        15127        15127  assemblies\System.Linq.dll
2021-02-23 16:48:20 .....       511674       511674  assemblies\System.Private.CoreLib.dll
2021-02-23 16:48:20 .....        63183        63183  assemblies\Java.Interop.dll
2021-02-23 16:48:20 .....        86450        86450  assemblies\Mono.Android.dll
2021-02-23 16:48:20 .....         6428         2139  lib\armeabi-v7a\libxamarin-app.so
2021-02-23 16:43:12 .....       722696       355325  lib\armeabi-v7a\libSystem.IO.Compression.Native.so
2021-02-23 16:43:12 .....        63276        30326  lib\armeabi-v7a\libSystem.Native.so
2021-02-23 16:43:12 .....        83736        33154  lib\armeabi-v7a\libSystem.Security.Cryptography.Native.OpenSsl.so
2021-02-23 16:43:12 .....      4246188      1741993  lib\armeabi-v7a\libmonosgen-2.0.so
2021-02-23 16:43:12 .....       195824        80689  lib\armeabi-v7a\libmonodroid.so
2021-02-23 16:43:12 .....        49688        12482  lib\armeabi-v7a\libxa-internal-api.so
2021-02-23 16:43:12 .....        25332        11144  lib\armeabi-v7a\libxamarin-debug-app-helper.so
------------------- ----- ------------ ------------  ------------------------
2021-02-23 16:48:20            6408833      3036233  20 files
```

I could have broken something though, let me know, thanks.

Reply:
`<_StrippedFrameworkNativeLibrary Include="@(FrameworkNativeLibrary->'$(IntermediateOutputPath)native\%(RuntimeIdentifier)\%(Filename)%(Extension)')" />`

So this did the trick. Thanks for looking into that and improving the task.

I have checked it with `BuildReleaseArm64` test and it still work as expected.

I have also removed the error message. The error message we get from `AndroidToolTask` contains all the information we need.

```
                     C:\Users\rodo\android-toolchain\dotnet\packs\Microsoft.Android.Sdk.win-x64\11.0.200-ci.master.121\tools\ndk\aarch64-linux-android-strip.exe --strip-d C:\Users\rodo\.nuget\packages\microsoft.netcore.app.runtime.android-arm64\6.0.0-preview.1.21102.12\runtimes\android-arm64\native\libSystem.IO.Compression.Native.so -o obj\Release\native\android-arm64\libSystem.IO.Compression.Native.so  (TaskId:300)
                     C:\Users\rodo\android-toolchain\dotnet\packs\Microsoft.Android.Sdk.win-x64\11.0.200-ci.master.121\tools\ndk\aarch64-linux-android-strip.exe: option `--strip-d' is ambiguous (TaskId:300)
                     Usage: C:\Users\rodo\android-toolchain\dotnet\packs\Microsoft.Android.Sdk.win-x64\11.0.200-ci.master.121\tools\ndk\aarch64-linux-android-strip.exe <option(s)> in-file(s) (TaskId:300)
                      Removes symbols and sections from files (TaskId:300)
                      The options are: (TaskId:300)
                       -I --input-target=<bfdname>      Assume input file is in format <bfdname> (TaskId:300)
                       -O --output-target=<bfdname>     Create an output file in format <bfdname> (TaskId:300)
                       -F --target=<bfdname>            Set both input and output format to <bfdname> (TaskId:300)
                       -p --preserve-dates              Copy modified/access timestamps to the output (TaskId:300)
                       -D --enable-deterministic-archives (TaskId:300)
                                                        Produce deterministic output when stripping archives (TaskId:300)
                       -U --disable-deterministic-archives (TaskId:300)
                                                        Disable -D behavior (default) (TaskId:300)
                       -R --remove-section=<name>       Also remove section <name> from the output (TaskId:300)
                       -s --strip-all                   Remove all symbol and relocation information (TaskId:300)
                       -g -S -d --strip-debug           Remove all debugging symbols & sections (TaskId:300)
                          --strip-dwo                   Remove all DWO sections (TaskId:300)
                          --strip-unneeded              Remove all symbols not needed by relocations (TaskId:300)
                          --only-keep-debug             Strip everything but the debug information (TaskId:300)
                       -N --strip-symbol=<name>         Do not copy symbol <name> (TaskId:300)
                       -K --keep-symbol=<name>          Do not strip symbol <name> (TaskId:300)
                          --keep-file-symbols           Do not strip file symbol(s) (TaskId:300)
                       -w --wildcard                    Permit wildcard in symbol comparison (TaskId:300)
                       -x --discard-all                 Remove all non-global symbols (TaskId:300)
                       -X --discard-locals              Remove any compiler-generated symbols (TaskId:300)
                       -v --verbose                     List all object files modified (TaskId:300)
                       -V --version                     Display this program's version number (TaskId:300)
                       -h --help                        Display this output (TaskId:300)
                          --info                        List object formats & architectures supported (TaskId:300)
                       -o <file>                        Place stripped output into <file> (TaskId:300)
                     C:\Users\rodo\android-toolchain\dotnet\packs\Microsoft.Android.Sdk.win-x64\11.0.200-ci.master.121\tools\ndk\aarch64-linux-android-strip.exe: supported targets: elf64-littleaarch64 elf64-bigaarch64 elf32-littleaarch64 elf32-bigaarch64 elf32-littlearm elf32-bigarm elf64-little elf64-big elf32-little elf32-big plugin srec symbolsrec verilog tekhex binary ihex (TaskId:300)
10:33:16.727   1:7>C:\Users\rodo\android-toolchain\dotnet\packs\Microsoft.Android.Sdk.win-x64\11.0.200-ci.master.121\targets\Microsoft.Android.Sdk.AssemblyResolution.targets(182,5): error MSB6006: "aarch64-linux-android-strip.exe" exited with code 1. [C:\Users\rodo\git\xa-today\bin\TestDebug\temp\BuildReleaseArm64False\UnnamedProject.csproj]
```